invenio_mail/tasks.py

```diff
@@ -79,4 +85,34 @@ def _send_email_with_attachments(data, attachments=[]):
             disposition = attachment.get("disposition")
         msg.attach(content_type=content_type, data=rawdata, disposition=disposition)
 
-    current_app.extensions["mail"].send(msg)
+    _send(msg, self)
+
+
+def _send(msg, task):
+    """Send emails and add exception handling."""
+    try:
+        current_app.extensions["mail"].send(msg)
+    except (
+        smtplib.SMTPRecipientsRefused,
+        smtplib.SMTPHeloError,
+        smtplib.SMTPSenderRefused,
+        smtplib.SMTPDataError,
+        smtplib.SMTPNotSupportedError,
+        Exception,
+    ) as excp:
+        if task.request.retries <= current_app.config["MAIL_MAX_RETRIES"]:
+            raise task.retry(
+                countdown=int(random.uniform(2, 4) ** task.request.retries)
+            )
+        else:
+            message = (
+                "Mail could not be dispatched!"
+                if isinstance(excp, smtplib.SMTPException)
+                else "Mail server does not answer to requests!"
+            )
+            current_app.logger.error(message)
+            if current_app.config["MAIL_LOG_FAILED_MESSAGES"]:
+                current_app.logger.info(msg.body)
+                if msg.attachments:
+                    for attachment in msg.attachments:
+                        current_app.logger.info(attachment.data)
```

Comment:
Sorry to come back to this... it looks to me quite a specific use case.
Can't you log some information instead that allows you to find what attachment fail (E.g. filename/checksum/date...)?

Otherwise, and instead of having a global config var `MAIL_LOG_FAILED_MESSAGES`, I suggest having a dedicate invenio-mail python logger, that you can configure in your instance, and use the logging levels for that logger.

E.g. (pseudocode)
```
logger = getLogger("invenio-mail-attachements")

...
logger.debug(attachement.data)
...
```
This has to be tested, as we have quite some trouble in the past tuning the logging levels: the `root` logger level needs to be configured first, see [here](https://github.com/inveniosoftware/invenio-logging/issues/67).

Reply:
I removed the logging part, internally it was a ticket of it's own. The attachment is nowhere/never saved, it's just a certificate for the user that is generated on the fly, so I think the dedicated logger is the better way.